File: amstrong.py
num=int(input("number"))
num_string=str(num)
num_digits=len(num_string)
sum=0
temp=num
while temp>0:
    digit=temp%10
    sum+=digit**num_digits
    temp//=10
if num == sum: 
    print("Armstrong") 
else: 
    print("Not Armstrong") 

File: amstrong.java
import java.util.Scanner;

public class amstrong {
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        String numstring=String.valueOf(n);
        int numdigits=numstring.length();
        int sum=0;
        for(char digitchar:numstring.toCharArray()){
            int digit=Character.getNumericValue(digitchar);
            sum+=Math.pow(digit,numdigits);
        }
        if(n==sum){
            System.out.println("amstrong");
        }else{
            System.out.println("not a amstrong");
        }
    }
}
